tortoise/verifying_algorithm.go

```diff
@@ -115,6 +122,7 @@ func (trtl *ThreadSafeVerifyingTortoise) BaseBlock(ctx context.Context) (types.B
 
 // HandleLateBlocks processes votes and goodness for late blocks (for late block definition see white paper).
 // Returns the old verified layer and new verified layer after taking into account the blocks' votes.
+// DEPRECATED: don't use this method it will be completely removed.
 func (trtl *ThreadSafeVerifyingTortoise) HandleLateBlocks(ctx context.Context, blocks []*types.Block) (types.LayerID, types.LayerID) {
```

Comment:
why not remove now?

Reply:
it was used in some healing tests, i could probably refactor that in this change, but would rather do it later

Reply:
but it seems that self-healing just needs the late blocks to be part of the mesh. i'd imagine adding the late blocks to the layer is relatively straightforward and can be done in this PR? if not as simple as i described, does that mean self-healing require some part of this code path?

Reply:
i don't think that self-healing can depend on HandleLateBlocks in any meaningful way that can work. also the test is weird, it hand-crafted list of blocks that it wants to process and it fails if it re-processes all of them. i would rather rewrite whole test with declarative setup instead of debugging this one

Reply:
i removed HandleLateBlocks